packages/runtime-core/__tests__/componentProxy.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { mockWarn } from '@vue/runtime-test'
+
+describe('component proxy', () => {
+  /**
+   * These tests should tests the warnings for PublicInstanceProxyHandlers in componentProxy
+   * {@link @vue/runtime-core/src/componentProxy}
+   *
+   * The function setupStatefulComponent is the location where PublicInstanceProxyHandlers is being used.
+   * {@link @vue/runtime-core/src/createRenderer}
+   *
+   * Currently I haven't found a way to create a component that uses PublicInstanceProxyHandlers
```

Comment:
The `PublicInstanceProxy` is simply `this` inside any Vue 2.x options :)

Reply:
Ok, done!